include/tdi/common/tdi_init.hpp

```diff
@@ -165,6 +189,9 @@ class DevMgr {
 
   tdi_status_t deviceRemove(const tdi_dev_id_t &device_id);
 
+  tdi_status_t deviceWarmInitBegin(const tdi_dev_id_t &device_id, const WarmInitOptions &warm_init_options);
```

Comment:
I think it is better this way so that the application can just call: devMgr->deviceWarmInitBegin
instead of devMgr->warmInitImpl()->deviceWarmInitBegin.

I'm thinking I will just get rid of the getter(warmInitImpl())